erigon-lib/state/domain.go

```diff
@@ -1196,11 +1197,13 @@ func (dt *DomainRoTx) Unwind(ctx context.Context, rwTx kv.RwTx, step, txNumUnwin
 
 		ic, err := dt.ht.IdxRange(k, int(txNumUnwindTo)-1, 0, order.Desc, -1, rwTx)
 		if err != nil {
+			ic.Close()
```

Comment:
better in `defer`

Reply:
ttl of that cursor is one cycle of the loop, so imo better to close that cursor right there

Reply:
@awskii move cycle body to a func/method. Also it's responsibility of `IdxRange` func to close iter if it returns error.